client-common/src/main/java/org/apache/cassandra/sidecar/common/ApiEndpointsV1.java

```diff
@@ -121,13 +121,16 @@ public final class ApiEndpointsV1
     public static final String LIST_CDC_SEGMENTS_ROUTE = API_V1 + CDC_PATH + "/segments";
     public static final String STREAM_CDC_SEGMENTS_ROUTE = LIST_CDC_SEGMENTS_ROUTE + "/" + SEGMENT_PATH_PARAM;
 
+    // Schema Reporting
+    private static final String REPORT_SCHEMA = "/report-schema";
+    public static final String REPORT_SCHEMA_ROUTE = API_V1 + REPORT_SCHEMA;
```

Comment:
I'm not 100% convinced about this route. What do you think about this?
```suggestion
    public static final String REPORT_SCHEMA_ROUTE = API_V1 + "/services/schema-reporting/ REPORT_SCHEMA;
```

or something that scopes this feature to the schema reporting service.

Reply:
I agree. The API should not contain verb. Given the API is to interact with DataHub, I would suggest the following to explicitly refer to DataHub

```
PUT /api/v1/datahub/schemas
```

Meaning, update the schema in datahub. 

The API on the datahub side is `/openapi/entities/v1`. Maybe `entities` also work if we want to define a similar API. However, I think `schemas` provides better clarity in the context of Sidecar.

Reply:
`schema-reporting` would map directly to our configuration in `sidecar.yaml`, so I'd prefer to have some consistency between what's user facing (sidecar.yaml) and the endpoints . WDYT?

Reply:
The `schema_reporting` in sidecar.yml is to configure the scheduled task, which is different from the API. 

IMO, `PUT .../schemas` means publishing to the `schemas` resource already. The `publishing` portion is redundant. Having the same term in the configuration and API is not a goal and might be impractical to achieve, since the API scheme and ymal are unrelated.

Reply:
I would agree that `PUT .../schema` (singular, not plural) is an appropriate name if we provide the schema we intent to persist. But we're not putting schema in. We're triggering an event that causes the schema to be internally produced and reported. Using `PUT .../schema-report` would probably follow the general idea of REST better. What do you think?

Reply:
In a RESTful design, the resource in the plural form is a more popular convention. The API is to publishing/inserting a schema entity in the Datahub's `entities` resource. The intent of publishing is expressed by the HTTP verb (`PUT`. TBH, it should be `POST`, but I am fine with `PUT`), and the receiver is the resource collection, i.e. `schemas`. `datahub` in the API is to clarify the external service.

As mentioned in my previous comment, in DataHub's term, they are `entities`. But in the consideration of the sidecar clients, it is clearer with calling them `schemas`.

Reply:
I get your point, can you specify the exact string you want me put in there, so there's no confusion.